### DISCUSSION TITLE
How to encode root/parent relationships in pollen growth data

### DISCUSSION THREAD

**Discussion by cedarwarman** [quantumjot/btrack#221] [Q&A]
Hi, I'm not sure if this question is a little too distant from the typical use case of cell-tracking, but I figured I would ask.

I am tracking pollen grains and pollen tube tips in images like this:
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/28878238/224182247-0876be8b-3d85-4ed9-b9d7-422a6b69d84d.png">

Right now I'm tracking tube tips and pollen grains separately, but I would like to associate each tube tip with a pollen grain (as the parent, if I'm using the terminology correctly). I'm wondering if there's a way to encode this behavior in BayesianTracker. Pollen behaves a little different than other cells, it would have to follow these rules:

- Each tube tip track needs to have 1 pollen grain parent
- Pollen grains can have up 3, but usually 1, tube growing out of them (child nodes)
- Pollen grains and tube tips are distinguished in the data by their class, which I've added to the `PyTrackObject` as an extra property. Pollen grains can be one of several classes (e.g. `ungerminated`, `germinated`, `burst`) while pollen tube tips are always the `pollen_tube_tip` class.

Thanks for your help!

**Comment by quantumjot** [comment]
Sorry I didn't get back to this - I think this particular question might be challenging to address with btrack because of the way it works. However, we have another project that might be interesting for this: https://github.com/alan-turing-institute/grace
